api/pkg/artifact/artifact.go

```diff
@@ -133,24 +176,135 @@ func (gac *GcsArtifactClient) DeleteArtifact(ctx context.Context, url string) er
 	return nil
 }
 
+type S3ArtifactClient struct {
+	URLScheme
+	client *s3.Client
+}
+
+func NewS3ArtifactClient() (Service, error) {
+	cfg, err := config.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("%s,failed loading s3 config for the artifact client", err.Error())
+	}
+	client := s3.NewFromConfig(cfg, func(o *s3.Options) {
+		o.BaseEndpoint = aws.String(os.Getenv("AWS_ENDPOINT_URL"))
```

Comment:
just realized this, why this is not part of the config? or being passed during initialization?

Reply:
Ah it's needed because we need to specify the base endpoint for the s3 client and that base endpoint isn't being read automatically as an env var by `config.LoadDefaultConfig`, unlike other env vars like `AWS_ACCESS_KEY_ID`, `AWS_DEFAULT_REGION` or `AWS_SECRET_ACCESS_KEY`.

Here's an example of how to specify the base endpoint manually in the s3 client: https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/#with-baseendpoint. Basically we need to manually pass it as an option function to `s3.NewFromConfig`.